src/main/java/seedu/planner/model/user/UserProfile.java

```diff
@@ -7,52 +7,44 @@
 
 import seedu.planner.model.course.FocusArea;
 import seedu.planner.model.course.Major;
-import seedu.planner.model.user.exceptions.UserProfileNotSetUpException;
 
 //@@author GabrielYik
 
 /**
  * Represents the profile of the user.
  */
 public class UserProfile {
+    private final int year;
+    private final int semester;
+    private final Major major;
+    private final List<FocusArea> focusAreas;
 
-    private static UserProfile instance;
-    private int year;
-    private int semester;
-    private Major major;
-    private List<FocusArea> focusAreas;
-
-    private UserProfile(int year, int semester, Major major, List<FocusArea> focusAreas) {
-        this.year = year;
-        this.semester = semester;
-        this.major = major;
-        this.focusAreas = focusAreas;
+    public UserProfile() {
+        this(1, 1, Major.COMPUTER_SCIENCE, new ArrayList<>());
     }
 
     /**
-     * Gets an instance of the user profile.
-     * If the user profile does not exist,
-     * {@code UserProfileNotSetUpException} is thrown.
-     *
-     * @return The user profile instance.
+     * Copy constructor
      */
-    public static UserProfile getInstance() {
-        if (instance == null) {
-            throw new UserProfileNotSetUpException();
-        }
-        return instance;
+    public UserProfile(UserProfile copy) {
+        this.year = copy.year;
```

Comment:
Can remove the `this`s

Reply:
Just my personal preference to use `this` in constructors. For other functions, I use `this` only if needed.